sim-lib/src/lib.rs

```diff
@@ -349,6 +349,19 @@ pub struct WriteResults {
     pub batch_size: u32,
 }
 
+/// PayGenDisplay is a marker trait to allow creation of trait objects for printable payment generators.
+pub trait PayGenDisplay: PaymentGenerator + Display {}
+
+/// ExecutorKit contains the components required to spin up an activity configured by the user, to be used to
+/// spin up the appropriate producers and consumers for the activity.
+struct ExecutorKit {
+    source_info: NodeInfo,
+    /// We use an arc mutex here because some implementations of the trait will be very expensive to clone.
+    /// See [NetworkGraphView] for details.
+    network_generator: Arc<Mutex<dyn DestinationGenerator + Send>>,
+    payment_generator: Box<dyn PayGenDisplay + Send>,
```

Comment:
We could also append `Send` declarations on traits that we dynamically dispatch, so `pub trait PaymentGenerator: Display + Send {}` means we get to declare `<dyn PaymentGenerator>` instead of `<dyn PayGenDisplay + Send>`

Reply:
👍  did this for `DestinationGeneartor` as well.